src/Models/ListModel.vala

```diff
@@ -35,6 +35,10 @@ public class Akira.Models.ListModel<Model> : GLib.Object, GLib.ListModel {
         return list.nth_data (position) as Object;
     }
 
+    public Model? nth_data (uint position) {
+        return list.nth_data (position);
+    }
```

Comment:
I noticed you tend to add a lot of these helper functions, but I'm not sure this is an optimal case.
Helper functions are necessary when dealing with repeated methods, conditions, or calculations, that need to be called in multiple locations throughout the application.

This `nth_data()` method basically only removes the necessity of calling the list object.
From `items.list.nth_data (items_length - 1 - i);` to `items.nth_data (items_length - 1 - i);`.
I don't see any benefit to this, and it can also be quite confusing for other devs or even ourselves in the future since this method is named exactly like the original method.

We should try to avoid adding helper functions without any login within.
What do you think?

Reply:
Yes, I tend to create helper methods, when they're needed. 

Here I totally agree with you that this thing isn't very useful. 

But it tries to solve a lot of quirks that happended to accumulate on this particular spot eheh. 

First, the list inside the ListModel is a private member and I think for a good reason, so I just don't want to expose it and use directly. 

Second, the `get_item` method in ListModel is required by the GLib.ListModel interface to return an GLib.Object and I didn't want to cast the item I get from the ListModel here.

The third issue is that it seems that Vala does not expose any bracket indexing API. I just wanted to the the i-th element of a list by using the [] operator, but, at a first glance, I've been able to find the way to overload the indexing operator in ListModel to provide this way of getting the items inside with the proper type. 

So I created this tiny wrapper around the list `nth_data`, just to accomplish this super simple task that Vala is not providing. 

I think I'll dig deeper trying to find a way to do this using only Vala, otherwise I'll simply use the `get_item` with casting!

Reply:
It has been as simple as that, I've just added these lines to the `ListModel` class:

```vala
    public Model get(uint index) {
        return list.nth_data (index);
    }
```

You simply need to add the `get` (and the corresponding `set` method) in order to enable the bracket indexing of the item itself! 

Super cool! 

I found it here, I think you might be interested: https://naaando.gitbooks.io/the-vala-tutorial/en/5-advanced-features/methods-with-syntax-support.html